### PR TITLE
Shortcode: add example give_raised_for_forms

### PR DESCRIPTION
**Pull request by matheuswd**
Creates a shortcode to display a message with the total number of donations for a specific donation form.

### CODE REVIEW

File: custom-shortcodes/donations-for-a-form-shortcode.php [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+	if ( get_post_type( $atts[ 'form_id' ] ) !== 'give_forms' ) {
+		$html = '
+			<div class="give_raised_for_forms">
+				<p>
+					This is not a donation form. Make sure you are using an ID for a donation form.
+				</p>
+			</div>
+		';
+
+		return $html;
```
Comment (jonwaldstein) [inline custom-shortcodes/donations-for-a-form-shortcode.php:47]: I'm seeing this html structure consistently for all your conditions.  The only difference is the content inside the `<p>` tag.  To make this whole function more [DRY](https://en.wikipedia.org/wiki/Don%27t_repeat_yourself) I would suggest doing all your conditionals inside one instance of the html.

something like this...
```
$atts = shortcode_atts(array(
    'form_id' => false,
), $atts, 'give_raised_for_forms');

$formId = $atts['form_id'];

if (!$formId) {
    $message = 'It\'s necessary that you insert the form ID on the shortcode.';
} elseif (get_post_type($formId) !== 'give_forms') {
    $message = 'This is not a donation form. Make sure you are using an ID for a donation form.';
} else {
    $form_title = get_the_title($formId);
    $number_of_donations = give_get_meta($formId, '_give_form_sales', true);

    $message = sprintf(
        'We already have %s donations for the %s campaign!',
        $number_of_donations,
        $form_title
    );
}

return '<div class="give_raised_for_forms"><p> ' . $message . '</p></div>';
```

File: custom-shortcodes/donations-for-a-form-shortcode.php [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+	}
+
+	$info = get_post_meta( $atts[ 'form_id' ] );
+	$post_title = get_the_title( $atts[ 'form_id' ] );
+
+	$number_of_donations = $info[ '_give_form_sales' ][0];
```
Comment (jonwaldstein) [inline custom-shortcodes/donations-for-a-form-shortcode.php:53]: You can simplify this with `$number_of_donations = give_get_meta($formId, '_give_form_sales', true);`

**Review by jonwaldstein — CHANGES_REQUESTED** [review]
Hey @matheuswd great first pass! This definitely works but I added some comments and suggestions to improve upon what you have.

File: custom-shortcodes/donations-for-a-form-shortcode.php [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+
+function my_give_display_donations_for_forms( $atts ) {
+
+	$atts = shortcode_atts( array(
+		'form_id' => false,
+	), $atts, 'give_raised_for_forms' );
+
+	$html = '';
+
+	if ( ! $atts[ 'form_id' ] ) {
```
[the remark this answers is not in the document]
Comment (matheuswd) [reply custom-shortcodes/donations-for-a-form-shortcode.php]: It makes total sense. Thanks

File: custom-shortcodes/donations-for-a-form-shortcode.php [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+		return $html;
+	}
+
+	$info = get_post_meta( $atts[ 'form_id' ] );
+	$post_title = get_the_title( $atts[ 'form_id' ] );
+
+	$number_of_donations = $info[ '_give_form_sales' ][0];
+
+	$html = '
+		<div class="give_total_forms">
```
[the remark this answers is not in the document]
Comment (matheuswd) [reply custom-shortcodes/donations-for-a-form-shortcode.php]: 👍

File: custom-shortcodes/donations-for-a-form-shortcode.php (hunk 1 above)
Comment (matheuswd) [reply custom-shortcodes/donations-for-a-form-shortcode.php]: Much better!

File: custom-shortcodes/donations-for-a-form-shortcode.php (hunk 2 above)
Comment (matheuswd) [reply custom-shortcodes/donations-for-a-form-shortcode.php]: thanks! I've never used this so it's learning opportunity.

**Comment by matheuswd** [comment]
@jonwaldstein I've reviewed everything and I think it's ok nw. Can you review it?

**Review by jonwaldstein — APPROVED** [review]
@matheuswd great job!

**Pull request merged**